Dockerfile

```diff
@@ -1,12 +1,15 @@
-FROM gcr.io/google.com/cloudsdktool/cloud-sdk:slim
+FROM gcr.io/google.com/cloudsdktool/cloud-sdk:alpine
 
-RUN apt-get update && \
-    apt-get install -y curl gnupg && \
-    echo "deb [signed-by=/usr/share/keyrings/cloud.google.gpg] http://packages.cloud.google.com/apt cloud-sdk main" | tee -a /etc/apt/sources.list.d/google-cloud-sdk.list && \
-    curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key --keyring /usr/share/keyrings/cloud.google.gpg  add - && \
-    apt-get update -y && \
-    apt-get install google-cloud-sdk-gke-gcloud-auth-plugin -y && \
-    apt-get -y install kubectl
+RUN apk --update add curl && \
+    apk add kubectl && \
+    gcloud components install gke-gcloud-auth-plugin && \
+    rm -rf google-cloud-sdk/platform && \
+    rm -rf google-cloud-sdk/bin/anthoscli && \
+    rm -rf google-cloud-sdk/.install && \
+    rm -rf google-cloud-sdk/data && \
+    rm -rf usr/local/libexec && \
+    rm -rf usr/local/bin/docker && \
+    rm /var/cache/apk/*
```

Comment:
Instead of removing all the stuff you don't need, considering moving just the stuff you need via multi stage.

Reply:
Moved towards the multi stage since it makes the image even smaller, updated 👍